lib/vintage_net_mobile/asu_calculator.ex

```diff
@@ -0,0 +1,75 @@
+defmodule VintageNetMobile.ASUCalculator do
+  @moduledoc """
+  Convert raw ASU values to friendlier units
+
+  See https://en.wikipedia.org/wiki/Mobile_phone_signal#ASU for
+  more information.
+
+  The following conversions are done:
+
+  * dBm
+  * Number of "bars" out of 4 bars
+  """
+
+  @typedoc """
+  Number of bars out of 4 to show in a UI
+  """
+  @type bars :: 0..4
+
+  @typedoc """
+  dBm
+  """
+  @type dbm :: neg_integer()
+
+  @typedoc """
+  GSM ASU values
+
+  ASU values map to RSSI. 99 means unknown
+  """
+  @type gsm_asu :: 0..31 | 99
+
+  @typedoc """
+  UMTS ASU values
+
+  ASU values map to RSCP
+  """
+  @type umts_asu :: 0..90 | 255
+
+  @typedoc """
+  LTE ASU values
+
+  ASU values map to RSRP
+
+  https://arimas.com/78-rsrp-and-rsrq-measurement-in-lte/
+  """
+  @type lte_asu :: 0..97
+
+  @spec from_gsm_asu(gsm_asu()) :: %{asu: gsm_asu(), dbm: dbm(), bars: bars()}
```

Comment:
Do we want to add `@doc` for this function since this is a public module?

Reply:
Thanks. Fixed.